### PR TITLE
Fixed metadata propagation in Dataframe.any

### PR DESCRIPTION
**Pull request by jiawei-zhang-a**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.
The goal here is to make progress on https://github.com/pandas-dev/pandas/issues/28283.

### CODE REVIEW

**Comment by mroeschke** [comment]
Looks like there's some typing failures

**Comment by jiawei-zhang-a** [comment]
Propogate only when it is Series and ignore the return-value typing error when it is bool

File: pandas/core/frame.py [hunk 1]
```diff
@@ -10958,9 +10958,14 @@ def any(  # type: ignore[override]
     ) -> Series:
         # error: Incompatible return value type (got "Union[Series, bool]",
         # expected "Series")
-        return self._logical_func(  # type: ignore[return-value]
+
+        result = self._logical_func(
             "any", nanops.nanany, axis, bool_only, skipna, **kwargs
         )
+        if isinstance(result, (np.bool_, bool)):
+            return result  # type: ignore[return-value]
```
Comment (twoertwein) [inline pandas/core/frame.py:10966]: Could add `bool | np.bool__` to the return types (but then you might need to add ignores elsewhere in pandas where `any()` is called)

Comment (jiawei-zhang-a) [reply pandas/core/frame.py:10966]: Thank you for your suggestion, let me try

Comment (twoertwein) [reply pandas/core/frame.py]: You probably need to add ignores here:

> pandas/core/reshape/encoding.py:458: error: Item "bool" of "Union[Series, bool_, bool]" has no attribute "any"  [union-attr]
> pandas/core/reshape/encoding.py:461: error: Item "bool_" of "Union[Series, bool_, bool]" has no attribute "idxmax"  [union-attr]
> pandas/core/reshape/encoding.py:461: error: Item "bool" of "Union[Series, bool_, bool]" has no attribute "idxmax"  [union-attr]

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**